matchbox/resource_profile.go

```diff
@@ -156,6 +163,19 @@ func resourceProfileDelete(d *schema.ResourceData, meta interface{}) error {
 	return nil
 }
 
-func containerLinuxConfigName(name string) string {
-	return name + ".yaml.tmpl"
+func containerLinuxConfigName(d *schema.ResourceData) string {
+	filename, _ := containerLinuxConfig(d)
+	return filename
+}
+
+func containerLinuxConfig(d *schema.ResourceData) (filename, config string) {
```

Comment:
Should error if neither format is provided. Also, you can probably simplify this,

```
name := d.Get("name").(string)
if content, ok := d.GetOk("raw_ignition"); ok {
    return fmt.Sprintf("%s.ign", name), content.(string)
}
...
```